integration/service/update_test.go

```diff
@@ -32,37 +32,37 @@ func TestServiceUpdateLabel(t *testing.T) {
 	service.Spec.Labels["foo"] = "bar"
 	_, err := cli.ServiceUpdate(ctx, serviceID, service.Version, service.Spec, types.ServiceUpdateOptions{})
 	assert.NilError(t, err)
-	poll.WaitOn(t, serviceIsUpdated(cli, serviceID), swarm.ServicePoll)
+	poll.WaitOn(t, serviceIsUpdated(cli, serviceID, true), swarm.ServicePoll)
```

Comment:
If I understand correctly; 

Nil is allowed if only the service's metadata is updated, but no tasks (containers) are re-created.

For example; these don't re-deploy a new container;

- `docker service update --update-delay=10s test`
- `docker service update --label-add=hello-world test`

But (interestingly, because `--reserve-memory` only updates accounting) this does;

- `docker service update --reserve-memory=400M test`


Two thoughts on this;

- Would it be clearer if we had separate `serviceSpecIsUpdated` and `serviceIsUpdated` functions, instead of a boolean (the `allowNil` doesn't really describe the "why")
- Is checking for `nil` the best way to check if it reconciled? 

Would it be an idea to use the `.Version.Index` as a check?

```bash
docker service create --name test nginx:alpine
docker service inspect --format '{{.Version.Index}}' test
121
docker service update --label-add=hello-world test

docker service inspect --format '{{.Version.Index}}' test
128
```

Reply:
Sounds reasonable. Updated